### ISSUE TITLE
Use of torch.empty when inputs are masked causes NaNs

### ISSUE THREAD

**Issue by jleben** [nasaharvest/galileo#15]
I have been getting NaNs when using the Galileo tiny encoder in eval mode, with official pretrained weights.

Upon detailed investigation, this is what I observe:
- My inputs to the Galileo never contain any inf or nan values
- In `apply_linear_projection`, `torch.empty` is used instead of the projection for masked tokens. ([link](https://github.com/nasaharvest/galileo/blob/0983eb226d7c55d63d0643cb2b248fec52a49e6c/src/galileo.py#L689)). This causes those masked tokens to have unpredictable values, sometimes infinity.
- In the standard attention mechanism, the masked tokens and values dependent on them are not entirely ignored (only multiplied with a very small value) and sometimes cause the output to be NaN.

Replacing `torch.empty` with `torch.zeros` in `apply_linear_projection` resolves the issue for me.

**Comment by gabrieltseng** [comment]
Thanks for sharing the example. The issue is with the following lines:

```python
# Half of the samples have a sequence length of 3
s_t_m[:50, :, :, 0:3, 0] = 0
# Other half of the samples have a sequence length of 4
s_t_m[50:, :, :, 0:4, 0] = 0
```

This only assigns the **first dimension** of s_t_m to be unmasked. However, Galileo assumes that all dimensions within a band-group have the same mask (since a band-group will be combined into a single token). The band groups are documented here: https://github.com/nasaharvest/galileo/blob/6375daa2bf31bbf1a4be8eb67705c610cb0e8c82/src/data/dataset.py#L65

The following two changes make the assertions pass:

1. Mask all the dimensions for a timestep consistently:

```python
# Half of the samples have a sequence length of 3
s_t_m[:50, :, :, 0:3, :] = 0
# Other half of the samples have a sequence length of 4
s_t_m[50:, :, :, 0:4, :] = 0
```

2. Mask a single band group consistently. The first two dimensions in `s_t_m` make up the S1 band-group:

```python
# Half of the samples have a sequence length of 3
s_t_m[:50, :, :, 0:3, :2] = 0
# Other half of the samples have a sequence length of 4
s_t_m[50:, :, :, 0:4, :2] = 0
```

I'll add a note in the README about this since this is definitely a gotcha.
